### PR TITLE
RISC-V: enable compiler tests in CI pipeline

### PR DESCRIPTION
**Pull request by janvrany**
This PR enables compiler tests on RISC-V native build CI pipelines. Tests known to fail are disabled and will be dealt with in subsequent PRs.

### CODE REVIEW

**Comment by Leonardo2718** [comment]
AppVeyor build failed due to infrastructural issues so I restarted it.

**Comment by janvrany** [comment]
@Leonardo2718 : Thanks, may I also kindly ask you to poke genie to rebuild on all other platforms?

**Comment by fjeremic** [comment]
FYI a lot of the changes here will conflict with #4811 and #4814 once they are merged as they fix all the problems on Z. I'd prefer if this PR was not merged before those.

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by janvrany** [comment]
@Leonardo2718: Thanks! 
@fjeremic: OK, let's wait until these two are merged. Once done, I'll rebase and resolve conflicts.

**Comment by janvrany** [comment]
ffa801a is rebased after merge of #4811 and #4814 . Is there anything else to improve?

File: fvtest/compilertriltest/IfxcmpgeReductionTest.cpp [hunk 1]
```diff
@@ -25,7 +25,8 @@
 
 // C++11 upgrade (Issue #1916).
 template <typename ValType>
-class IfxcmpgeReductionTest : public ::testing::TestWithParam<std::tuple<ValType, ValType, int32_t (*)(ValType, ValType)>>
+//class IfxcmpgeReductionTest : public ::testing::TestWithParam<std::tuple<ValType, ValType, int32_t (*)(ValType, ValType)>>
+class IfxcmpgeReductionTest : public TRTest::TestWithPortLib, public ::testing::WithParamInterface<std::tuple<ValType, ValType, int32_t (*)(ValType, ValType)>>
```
Comment (fjeremic) [inline fvtest/compilertriltest/IfxcmpgeReductionTest.cpp]: Can we remove the comment here?

File: fvtest/jitbuildertest/CMakeLists.txt [hunk 2]
```diff
@@ -56,4 +56,8 @@ target_link_libraries(jitbuildertest
 
 set_property(TARGET jitbuildertest PROPERTY FOLDER fvtest)
 
-add_test(NAME JitBuilderTest COMMAND jitbuildertest --gtest_output=xml:${CMAKE_CURRENT_BINARY_DIR}/jitbuildertest-results.xml)
+# Disable JitBuilder tests on RISC-V. This is essentially a workaround
+# until #4764 is addressed. Once done, tests should be properly skipped.
```
Comment (fjeremic) [inline fvtest/jitbuildertest/CMakeLists.txt]: Tagging #4764 here so GitHub links the issue here.

File: fvtest/compilertriltest/IfxcmpgeReductionTest.cpp (hunk 1 above)
Comment (janvrany) [reply fvtest/compilertriltest/IfxcmpgeReductionTest.cpp]: Oops, sure. Removed in f526369.

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by janvrany** [comment]
The failure on AIX / PPC64 seems to be caused by some Jenkins hiccup rather than PR itself:
```
16:57:15  27/27 Test #27: compunittest ......................   Passed    1.58 sec
16:57:15  
16:57:15  100% tests passed, 0 tests failed out of 27
16:57:15  
16:57:15  Total Test time (real) = 286.62 sec
```

**Pull request merged**